mxcubeweb/core/components/harvester.py

```diff
@@ -0,0 +1,505 @@
+                },
+                {
+                    "crystal_uuid": "46aca4a4-0c00-4685-b1e1-53071bdcb66d",
+                    "name": "TEST4",
+                    "state": "harvested",
+                    "acronym": "cryoprotectant",
+                    "img_url": "https://htxlab.embl.org//rawimages/2021//CD032401/6/FORMULATRIX_CD032401_6_08-09-2021_06_04_01_00_99_Vis.jpg",
+                    "img_target_x": 38.632400075415,
+                    "img_target_y": 41.762286324786,
+                },
+            ]
+
+        return crystal_list
+
+    def get_global_state(self):
+        try:
+            return HWR.beamline.harvester_maintenance.get_global_state()
+        except Exception:
+            return "OFFLINE", "OFFLINE", "OFFLINE"
+
+    def send_data_collection_info_to_crims(self) -> bool:
+        """Send Data collected to CRIMS
+
+        Return (bool): Whether the request failed (false) or not (true)
+        """
+        dataCollectionGroupId = ""
+        crystal_uuid = ""
+        harvester_device = HWR.beamline.harvester
+
+        try:
+            rest_token = HWR.beamline.lims.lims_rest.get_rest_token()
+            proposal = HWR.beamline.session.get_proposal()
+
+            crims_url = harvester_device.crims_upload_url
+
+            queue_entries = HWR.beamline.queue_model.get_all_dc_queue_entries()
+            dc_id = ""
+            for qe in queue_entries:
+                dataCollectionGroupId = qe.get_data_model().lims_group_id
+                crystal_uuid = (
+                    qe.get_data_model().get_sample_node().crystals[0].crystal_uuid
+                )
+                dc_id = qe.get_data_model().id
+
+                Crims.send_data_collection_info_to_crims(
+                    crims_url,
+                    crystal_uuid,
+                    dataCollectionGroupId,
+                    dc_id,
+                    proposal,
+                    rest_token,
+                )
+            return True
+        except Exception as ex:
+            msg = "get all queue entries failed, reason:  %s" % str(ex)
+            logging.getLogger("user_level_log").exception(msg)
+            return False
+
+    def get_calibration_state(self) -> bool:
+        return HWR.beamline.harvester.calibrate_state
+
+    def calibrate_pin(self) -> bool:
+        """
+            Pin Calibration Procedure
+            In other for the Centring procedure to work on a Harvested Pin
+            a Calibration Procedure need to be execute
+
+        Return (bool): whether the calibration procedure goes to end (True)
+        or had and exception (False)
+        """
+        harvester_device = HWR.beamline.harvester
+
+        harvester_device.load_calibrated_pin()
+        harvester_device._wait_sample_transfer_ready(None)
+        print("waiting 40 seconds before mount")
+        # For some reason the Harvester return READY too soon
+        # approximately 40 Second sooner
+        gevent.sleep(40)
```

Comment:
Why no proper logging?

Isn't there any condition that we can check to make sure we have waited long enough?

Why 40 seconds specifically? Is it empirical?

Should the 40 seconds value be made configurable? Or at the very least extracted into a named constant?